### PR TITLE
39 web preview app view

### LINKED ISSUE #39: Web: Preview app-view

### PR DESCRIPTION
**Pull request by Aagaard00**

### CODE REVIEW

**Review by Gormiee — COMMENTED** [review]
LGTM. Svært at merge når jeg har ændret i alle Componenterne i Mobile.Components. Ændret meget i input field og button

File: src/UICrafter.Client/Components/AppViewDesigner.razor [hunk 1]
```diff
@@ -89,4 +91,15 @@
 
         component.Item.DropZoneID = component.DropzoneIdentifier;
     }
+
+    private async Task Preview()
```
Comment (Nephim) [inline src/UICrafter.Client/Components/AppViewDesigner.razor:95]: I am more a fan of doing preview as a tab at the top with the editor but this is fine for now.

File: src/UICrafter.Client/Components/Dialogs/AppViewPreviewDialog.razor [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+@code {
+    [CascadingParameter]
+    private MudDialogInstance MudDialog { get; set; } = null!;
+
+    [Parameter]
+    public required RepeatedField<UIComponent> UIComponents { get; set; }
+
+    private JsonPublisher stupidStore = new JsonPublisher();
```
Comment (Nephim) [inline src/UICrafter.Client/Components/Dialogs/AppViewPreviewDialog.razor]: stupidStore naming

File: src/UICrafter.Client/UICrafter.Client.csproj [hunk 3]
```diff
@@ -23,6 +23,9 @@
   </ItemGroup>
 
   <ItemGroup>
+    <Content Update="Components\Dialogs\AppViewPreviewDialog.razor">
```
Comment (Nephim) [inline src/UICrafter.Client/UICrafter.Client.csproj]: What is this?

File: src/UICrafter.Core/Components/BaseComponents/BaseLoadMetaDataList.razor [hunk 4]
```diff
@@ -12,8 +12,9 @@ else
     }
     else
     {
+        <MudTextField @bind-Value="SearchString" Class="ma-2" Placeholder="Search" Adornment="Adornment.Start" Immediate="true" AdornmentIcon="@Icons.Material.Filled.Search" IconSize="Size.Medium"></MudTextField>
         <MudList T="AppView">
-            @foreach (var view in AppViews)
+            @foreach (var view in AppViews.Where(x => x.Name.ToLower().Contains(SearchString.ToLower())))
```
Comment (Nephim) [inline src/UICrafter.Core/Components/BaseComponents/BaseLoadMetaDataList.razor]: You should do Contains(SearchString, InvariantCulture.IgnoreCase), or something along those lines.

File: src/UICrafter.Core/Components/Render/RenderAPIButton.razor [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+@{
+    base.BuildRenderTree(__builder);
+}
+
+@code {
+    [Parameter]
+    public required UIComponent Button { get; set; }
+
+    [CascadingParameter]
+    public required JsonPublisher StupidStore { get; set; }
```
Comment (Nephim) [inline src/UICrafter.Core/Components/Render/RenderAPIButton.razor]: Naming

File: src/UICrafter.Core/Components/Render/RenderAPIButton.razor [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            }
+            else
+            {
+                // Handle unsuccessful response (e.g., error status code)
+                ButtonColor = Color.Error;
+                _processing = false;
+
+                // Optionally log or handle error information
+                var errorMessage = await response.Content.ReadAsStringAsync();
+                Console.WriteLine($"Error: {errorMessage}");
```
Comment (Nephim) [inline src/UICrafter.Core/Components/Render/RenderAPIButton.razor:95]: Log

File: src/UICrafter.Core/Components/Render/RenderAPIButton.razor [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            }
+        }
+        catch (Exception ex)
+        {
+            // Handle exceptions (e.g., network issues, deserialization errors)
+            ButtonColor = Color.Error;
+            _processing = false;
+            Console.WriteLine($"Exception: {ex.Message}");
```
Comment (Nephim) [inline src/UICrafter.Core/Components/Render/RenderAPIButton.razor:103]: Log

File: src/UICrafter.Core/Components/Render/RenderInputField.razor [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+@{
+    base.BuildRenderTree(__builder);
+}
+
+@code {
+    [Parameter]
+    public required UIComponent InputField { get; set; }
+
+    [CascadingParameter]
+    public required JsonPublisher StupidStore { get; set; }
```
Comment (Nephim) [inline src/UICrafter.Core/Components/Render/RenderInputField.razor]: Naming

File: src/UICrafter.Core/Components/Render/RenderTextBox.razor [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+@{
+    base.BuildRenderTree(__builder);
+}
+
+@code {
+    [Parameter]
+    public required UIComponent Textbox { get; set; }
+
+    [CascadingParameter]
+    public required JsonPublisher StupidStore { get; set; }
```
Comment (Nephim) [inline src/UICrafter.Core/Components/Render/RenderTextBox.razor]: Naming

**Review by Nephim — CHANGES_REQUESTED** [review]
A few things

File: src/UICrafter.Client/UICrafter.Client.csproj (hunk 3 above)
Comment (Aagaard00) [reply src/UICrafter.Client/UICrafter.Client.csproj]: No clue, det er kommet af sig selv

File: src/UICrafter.Core/Components/Render/RenderAPIButton.razor (hunk 6 above)
Comment (Aagaard00) [reply src/UICrafter.Core/Components/Render/RenderAPIButton.razor:95]: Mathman stuff

File: src/UICrafter.Core/Components/Render/RenderAPIButton.razor (hunk 7 above)
Comment (Aagaard00) [reply src/UICrafter.Core/Components/Render/RenderAPIButton.razor:103]: Mathman stuff

File: src/UICrafter.Core/Components/Render/RenderInputField.razor (hunk 8 above)
Comment (Aagaard00) [reply src/UICrafter.Core/Components/Render/RenderInputField.razor]: Mathman stuff

File: src/UICrafter.Core/Components/Render/RenderTextBox.razor (hunk 9 above)
Comment (Aagaard00) [reply src/UICrafter.Core/Components/Render/RenderTextBox.razor]: Mathman stuff

File: src/UICrafter.Core/Components/Render/RenderAPIButton.razor (hunk 5 above)
Comment (Aagaard00) [reply src/UICrafter.Core/Components/Render/RenderAPIButton.razor]: Mathman stuff

**Pull request merged**